segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java

```diff
@@ -689,28 +690,32 @@ private CompletableFuture<Void> validateSystemSnapshotExistsInTxn(MetadataTransa
         if (null == systemSnapshot) {
             return CompletableFuture.completedFuture(null);
         }
-
+        val iterator = systemSnapshot.getSegmentSnapshotRecords().iterator();
         // For each segment in snapshot
         return Futures.loop(
-                systemSnapshot.getSegmentSnapshotRecords(),
-                segmentSnapshot ->
-                    txn.get(segmentSnapshot.segmentMetadata.getKey())
-                        .thenComposeAsync(m -> validateChunksInSegmentSnapshot(txn, segmentSnapshot), executor)
-                        .thenComposeAsync(vv -> validateSegment(txn, segmentSnapshot.segmentMetadata.getKey()), executor)
-                        .thenApplyAsync(v -> true, executor),
-                executor);
+                () -> iterator.hasNext(),
+                () -> {
+                        synchronized (systemSnapshot.getSegmentSnapshotRecords()) {
```

Comment:
This is non-local reasoning. If SystemSnapshotRecord needs to have some sort of synchronization that class needs to enforce it. Here SystemSnapshotRecord doesn't even specify what type the records are beyond Collection. So even it doesn't even know if it needs to synchronize. I can only assume that this block means that it's being instantiated somewhere with a non-threadsafe collection but I don't actually see that anywhere in this file. Is this line redundant? Or is it being passed as a something other than a vector in some other file?

Reply:
This line is redundant. I forgot to remove it :(